src/main/java/jenkins/plugins/logstash/persistence/ElasticSearchDao.java

```diff
@@ -80,12 +79,13 @@ final void init(String host, int port, String key, String username, String passw
       throw new IllegalArgumentException("host field must specify scheme, such as 'http://'");
     }
 
-    auth = null;
     if (StringUtils.isNotBlank(username)) {
       auth = Base64.encodeBase64String((username + ":" + StringUtils.defaultString(password)).getBytes());
+    } else {
+      auth = null;
     }
 
-    clientBuilder = HttpClientBuilder.create();
+    clientBuilder = factory == null ? HttpClientBuilder.create() : factory;
```

Comment:
This allows us to pass this field in for mocking or just initialize normally.  This makes for the least difference between mocking and normal behavior.